cmd/ketch/pool_remove.go

```diff
@@ -35,14 +36,28 @@ type poolRemoveOptions struct {
 }
 
 func poolRemove(ctx context.Context, cfg config, options poolRemoveOptions, out io.Writer) error {
```

Comment:
can we have a test for it?

Reply:
While I did write a test, I did not include it here.  The [fake client](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client/fake#ClientBuilder) included with Kubebuilder does not alter cluster namespace when creating a pool, so `poolRemove` fails with `failed to get namespace`.  I had a test that ignores this error, but it defeats the purpose of testing IMO so I left it out.

I did come across [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest#Environment.Start) which may enable us to test for namespace removal.  How do we feel about bringing this into our test suite? It is quite a bit more involved, setting up a full test environment + API Server.

With that said, I did test these changes locally and getting the namespace name from the pool lookup to remove the namespace works as expected, terminating the namespace associated with the pool-to-be-removed.

Reply:
Could you please try k8s.io/client-go/kubernetes/fake ?

Reply:
I already wrote a test file using fake, and when it runs,  `poolRemove` fails with `failed to get namespace` because `fake` doesn't create a namespace when the test creates a pool.  I can push the file to this PR if you want to take a look.

Reply:
@itsksaurabh test has been fixed with commit [47fe2f12](https://github.com/shipa-corp/ketch/pull/43/commits/47fe2f12b63853544a9288302ec1443b77f72de6)